readers/api/transport.go

```diff
@@ -71,67 +72,67 @@ func MakeHandler(svc readers.MessageRepository, tc tpolicies.ThingsServiceClient
 func decodeList(_ context.Context, r *http.Request) (interface{}, error) {
 	offset, err := apiutil.ReadUintQuery(r, offsetKey, defOffset)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, apiutil.ErrValidation)
```

Comment:
use `apiutil.ErrValidation` as the wrapper for this whole file

Reply:
Do they have to be different for enpoints and for transport? if not errValidation as the wrapper they all have err as the wrapper?

Reply:
It was to mainftain uniformity in all transport.go files, and in all endpoint.go files
as some functions returned error directly, and others were helper functions which were called before, so this arrangement looked easier to maintain

Reply:
okay, noted

Reply:
Is this absolutely necessary? If necessary for error encoding, we can even use `errors.Unwrap` to avoid this.